Integrations/Snowflake/Snowflake.py

```diff
@@ -0,0 +1,392 @@
+ROLE = PARAMS.get('role')
+PROXY = PARAMS.get('proxy', False)
+# How much time before the first fetch to retrieve incidents
+IS_FETCH = PARAMS.get('isFetch')
+FETCH_TIME = PARAMS.get('fetch_time')
+FETCH_QUERY = PARAMS.get('fetch_query')
+DATETIME_COLUMN = PARAMS.get('datetime_column')
+INCIDENT_NAME_COLUMN = PARAMS.get('incident_name_column')
+MAX_ROWS = 10000
+
+'''SETUP'''
+
+if not PROXY:
+    del os.environ['HTTP_PROXY']
+    del os.environ['HTTPS_PROXY']
+    del os.environ['http_proxy']
+    del os.environ['https_proxy']
+
+if IS_FETCH and not (FETCH_QUERY and DATETIME_COLUMN):
+    err_msg = 'When fetching is enabled there are two additional parameters that are required;'
+    err_msg += ' The fetch query that determines what data to fetch and the name of the column'
+    err_msg += ' in the fetched data that contains a datetime object or timestamp.'
+    return_error(err_msg)
+
+
+'''HELPER CLASSES'''
+
+
+class DemistoError(Exception):
+    """Basic exception for errors raised by Demisto"""
+    def __init__(self, msg=None):
+        if not msg:
+            msg = 'A Demisto integration error occurred'
+        super(DemistoError, self).__init__(msg)
+
+
+class FetchError(DemistoError):
+    """For errors that occur in the inside an integration's fetch_incidents method"""
+    def __init__(self, msg=None):
+        if not msg:
+            msg = 'An error occurred whilst trying to fetch incidents'
+        super(FetchError, self).__init__(msg)
+
+
+'''HELPER FUNCTIONS'''
+
+
+def convert_datetime_to_string(v):
+    """
+    Parses date, time, timedelta, or datetime object into string
+
+    parameter: (datetime/date/time/timedelta) v
+        The datetime/date/time/timedelta object to convert
+
+    returns:
+        Formatted string of the object
+    """
+    if isinstance(v, datetime):
+        return v.strftime('%Y-%m-%d %H:%M:%S.%f %z')
+    elif isinstance(v, date):
+        return v.strftime('%Y-%m-%d')
+    elif isinstance(v, dttime) or isinstance(v, timedelta):
+        return v.strftime('%H:%M:%S.%f')
+    return v
+
+
+def set_provided(params, key, val1, val2=None):
+    """
+    If value is provided, set it in the dict
+    """
+    if val1:
+        params[key] = val1
+    elif val2:
+        params[key] = val2
+
+
+def process_table_row(row, checks):
+    """
+    Check row data and reformat if necessary
+
+    parameter: (dict) row
+        The data (table row) that needs to be processed
+
+    parameter: (dict[str, list]) checks
+        Dictionary where the key is a string indicative of the type (or bucket of types) that needs
+        reformatting and the values are a list of column names whose data is of that type
+
+    returns:
+        Reformatted Row
+    """
+    for column_name, val in row.items():
+        if column_name in checks.get('isDecimal', []):
+            # Then check the value and reformat it if necessary
+            if type(val) == Decimal:
+                row[column_name] = str(val)
```

Comment:
is this safe casting ?

Reply:
I was worried about loss of precision when casting to a float...